core/src/main/java/org/elasticsearch/search/aggregations/metrics/geocentroid/InternalGeoCentroid.java

```diff
@@ -145,14 +146,27 @@ public Object getProperty(List<String> path) {
     }
 
     static class Fields {
-        static final String CENTROID = "location";
+        static final ParseField CENTROID = new ParseField("location");
+        static final ParseField COUNT = new ParseField("count");
+        static final ParseField CENTROID_LAT = new ParseField("lat");
+        static final ParseField CENTROID_LON = new ParseField("lon");
     }
 
     @Override
     public XContentBuilder doXContentBody(XContentBuilder builder, Params params) throws IOException {
+        return renderXContent(builder, params, centroid, count);
+    }
+
+    static XContentBuilder renderXContent(XContentBuilder builder, Params params, GeoPoint centroid, long count) throws IOException {
```

Comment:
No, we don't. I tried this in one PR by sharing something similar in the interface but that isn't such a good solution either. In former cases the two toXContent were slightliy different, in this case they are really similar, thats why I didn't want to copy them. But I get the point about dependencies and can copy the toXContent over if you prefer. 
One other note on this: we often share ParseFields or at least the field keys between InternalX and ParsedX, should we also already start pulling them out, e.g. into the interfaces? Maybe better later when we eventually do the splitting at some point in the future?